### PR TITLE
disttask: add integration test for resource control

### PR DESCRIPTION
**Pull request by D3Hunter**
### What problem does this PR solve?

Issue Number: ref #49008

Problem Summary:

### What changed and how does it work?
-  add integration test for resource control
- some refactor for existing cases
### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/disttask/framework/integrationtests/framework_ha_test.go [hunk 1]
```diff
@@ -82,24 +83,24 @@ func TestHARandomShutdownInDifferentStep(t *testing.T) {
 	submitTaskAndCheckSuccessForHA(c.Ctx, t, "😊", c.TestContext)
 }
 
-func TestHAReplacedButRunning(t *testing.T) {
```
Comment (D3Hunter) [inline pkg/disttask/framework/integrationtests/framework_ha_test.go:85]: those cases depends on `mockTiDBPartitionThenResume` which only remove the node from "PD" and pause execution temporaryly, no network is partitioned, and there's no multiple owner, so remove them

File: pkg/disttask/framework/testutil/table_util.go [hunk 2]
```diff
@@ -175,8 +175,10 @@ func GetSubtaskEndTime(ctx context.Context, mgr *storage.TaskManager, subtaskID
 
 // GetSubtaskNodes gets subtasks running nodes for one task for test.
 func GetSubtaskNodes(ctx context.Context, mgr *storage.TaskManager, taskID int64) ([]string, error) {
-	rs, err := mgr.ExecuteSQLWithNewSession(ctx,
-		`select distinct(exec_id) from mysql.tidb_background_subtask where task_key=%?`, taskID)
+	rs, err := mgr.ExecuteSQLWithNewSession(ctx, `
+		select distinct(exec_id) from mysql.tidb_background_subtask where task_key=%?
+		union
+		select distinct(exec_id) from mysql.tidb_background_subtask_history where task_key=%?`, taskID, taskID)
```
Comment (lance6716) [inline pkg/disttask/framework/testutil/table_util.go:181]: Do you need a DISTINCT over UNION?

File: pkg/disttask/framework/testutil/table_util.go [hunk 3]
```diff
@@ -175,8 +175,10 @@ func GetSubtaskEndTime(ctx context.Context, mgr *storage.TaskManager, subtaskID
 
 // GetSubtaskNodes gets subtasks running nodes for one task for test.
```
Comment (lance6716) [inline pkg/disttask/framework/testutil/table_util.go:176]: the comment should also be updated

**Review by lance6716 — COMMENTED** [review]
17/21
will revert later

File: pkg/disttask/framework/testutil/table_util.go (hunk 2 above)
Comment (D3Hunter) [reply pkg/disttask/framework/testutil/table_util.go:181]: subtasks can only be in `tidb_background_subtask` or it's history

File: pkg/disttask/framework/testutil/table_util.go (hunk 3 above)
Comment (D3Hunter) [reply pkg/disttask/framework/testutil/table_util.go:176]: seems also fit, what you mean?

Comment (lance6716) [reply pkg/disttask/framework/testutil/table_util.go:176]: I see the data is also queried from `tidb_background_subtask_history` so maybe

```suggestion
// GetSubtaskNodes gets nodes that are running or have run the task for test.
```

**Review by lance6716 — APPROVED** [review]

**Review by ywqzzy — APPROVED** [review]

**Pull request merged**